lang/LangPrimSource/PyrArrayPrimitives.cpp

```diff
@@ -2042,22 +2046,27 @@ int prArrayLace(struct VMGlobals *g, int numArgsPushed)
 				if (isKindOf(obj3, class_list)) {
 					obj3 = slotRawObject(&obj3->slots[0]); // get the list's array
 				}
-				if (obj3 && isKindOf(obj3, class_array)) {
+				if (obj3 && isKindOf(obj3, class_array) && obj3->size>0) {
 					m = j % obj3->size;
 					slotCopy(&obj2->slots[i],&obj3->slots[m]);
 				} else {
-					slotCopy(&obj2->slots[i],&slots[k]);
+					SetNil(&obj2->slots[i]);
```

Comment:
I'm not sure if this is correct. Let's say you have an object that is neither a list or an array. Won't this be replaced by `nil` then, instead of copied?

Reply:
That's an interesting point, and I've modified the code to narrow the case for SetNil to specifically only zero-length arrays. Worth considering that the way the code was originally written, and will still be after this revision, is that for objects that are neither lists or arrays we don't take any action - no copy or creation. So does that mean that a `nil` is likely to be in that spot anyway, as it is an uninitialized spot in a new array?